synapse/app/generic_worker.py

```diff
@@ -132,107 +129,12 @@
 from synapse.storage.databases.main.ui_auth import UIAuthWorkerStore
 from synapse.storage.databases.main.user_directory import UserDirectoryStore
 from synapse.storage.databases.main.user_erasure_store import UserErasureWorkerStore
-from synapse.types import JsonDict
 from synapse.util import SYNAPSE_VERSION
 from synapse.util.httpresourcetree import create_resource_tree
 
 logger = logging.getLogger("synapse.app.generic_worker")
 
 
-class KeyUploadServlet(RestServlet):
-    """An implementation of the `KeyUploadServlet` that responds to read only
-    requests, but otherwise proxies through to the master instance.
-    """
-
-    PATTERNS = client_patterns("/keys/upload(/(?P<device_id>[^/]+))?$")
-
-    def __init__(self, hs: HomeServer):
-        """
-        Args:
-            hs: server
-        """
-        super().__init__()
-        self.auth = hs.get_auth()
-        self.store = hs.get_datastores().main
-        self.http_client = hs.get_simple_http_client()
-        self.main_uri = hs.config.worker.worker_main_http_uri
-
-    async def on_POST(
```

Comment:
AFAICS: these behaviours are handled by the other KeyUploadServlet:

- warning when client requests a specific device_id
- 400 error if there is no device_id provided nor one from the requester

and I am assuming(?) that the `make_client(hs)` machinery handles `X-Forwarded-*` headers.

Reply:
That's what it looked like to me too, I thought for sure I'd have to deal with catching errors but it seemed ok. Would love to have other eyes verify that though. Also, no `X-Forwarded-*` headers necessary since it's not actually "proxying" anything in that sense(in technicality if not in spirit, anyways)

Reply:
Here's the 400 it returns
https://github.com/matrix-org/synapse/blob/1eed795fc56d95df3968e37f3a4db92f24513e15/synapse/rest/client/keys.py#L107-L110
It appears to log a warning when "Client uploading keys for a different device"